ext/php8/ddtrace.c

```diff
@@ -1249,6 +1325,50 @@ static PHP_FUNCTION(dd_trace_peek_span_id) {
     return_span_id(return_value, ddtrace_peek_span_id());
 }
 
+/* {{{ proto string dd_trace_peek_span() */
+static PHP_FUNCTION(dd_trace_peek_span) {
+    UNUSED(execute_data);
+    if (DDTRACE_G(open_spans_top)) {
+        RETURN_OBJ_COPY(&DDTRACE_G(open_spans_top)->span.std);
+    }
+    RETURN_NULL();
+}
+
+/* {{{ proto string DDTrace\start_span() */
+static PHP_FUNCTION(start_span) {
+    UNUSED(execute_data);
+    ddtrace_span_fci *span_fci = ddtrace_init_span();
+    ddtrace_open_span(span_fci);
+    RETURN_OBJ_COPY(&span_fci->span.std);
+}
+
+/* {{{ proto string DDTrace\close_span() */
+static PHP_FUNCTION(close_span) {
+    UNUSED(execute_data);
+    double finish_time_seconds = 0;
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "|f", &finish_time_seconds) != SUCCESS) {
+        ddtrace_log_debug("unexpected parameter. the environment variable name must be provided");
+        RETURN_FALSE;
+    }
+
+    // TODO: also prevent closing internal root spans? Would need marker though
+    if (!DDTRACE_G(open_spans_top) || !DDTRACE_G(open_spans_top)->execute_data) {
+        zend_throw_exception(NULL, "There is no user-span on the top of the stack. Cannot close.", 0);
+        RETURN_THROWS();
```

Comment:
Same as above, let's not throw. The maximum that we can accept (but we should avoid it) is that tracing is not working as expected, but users' requests never have to be interrupted because of us.